libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -141,13 +145,13 @@ export class CartService {
       couponCode: args.promoCode,
     });
 
-    return cart;
+    return { ...cart, metricsOptedOut };
   }
 
   /**
    * Create a new cart with the contents of an existing cart, in the initial state.
    */
-  async restartCart(cartId: string): Promise<ResultCart> {
+  async restartCart(cartId: string): Promise<WithMetricsFlagCart> {
```

Comment:
Could we revert the changes to `restartCart` and `setupCart`? 

It's not necessary to include `metricsOptedOut` as part of these methods, since they don't get used in the location these methods are called.

Not for this PR, but I'm actually tempted to change the return type of these methods to just `cart.id`, since that's the only value required by consumers of this method.

Reply:
sounds good